scapy/arch/bpf/core.py

```diff
@@ -182,8 +183,9 @@ def get_working_ifaces():
             os.close(fd)
 
     # Sort to mimic pcap_findalldevs() order
-    interfaces.sort(lambda ifname_left_and_ifid_left,
-                        ifname_right_and_ifid_right: ifname_left_and_ifid_left[1]-ifname_right_and_ifid_right[1])
+    interfaces.sort(key=cmp_to_key(lambda ifname_left_and_ifid_left, ifname_right_and_ifid_right:
```

Comment:
I don't think you need that function here! `interfaces.sort(key=lambda elt: elt[1]` should work.